http_client.go

```diff
@@ -110,50 +114,60 @@ func (c *httpClient) Delete(url string, headers http.Header) (Response, error) {
 
 	request.Header = headers
 
-	return c.do(request)
+	res, err := c.Do(request)
+	return toHeimdallResponse(res, err)
 }
 
-func (c *httpClient) do(request *http.Request) (Response, error) {
-	hr := Response{}
+func toHeimdallResponse(response *http.Response, err error) (Response, error) {
+	fmt.Println(response, err)
+	var hr Response
+	merr := valkyrie.NewMultiError()
+	if err != nil {
+		merr.Push(err.Error())
+	}
+	if response == nil {
+		return hr, merr.HasError()
+	}
+	if response.Body != nil {
+		hr.body, err = ioutil.ReadAll(response.Body)
+		if err != nil {
+			merr.Push(err.Error())
+			return hr, merr.HasError()
+		}
+		response.Body.Close()
+	}
+	hr.statusCode = response.StatusCode
+	return hr, merr.HasError()
+}
 
+// Do makes an HTTP request with the native `http.Do` interface
+func (c *httpClient) Do(request *http.Request) (*http.Response, error) {
 	request.Close = true
 	multiErr := valkyrie.NewMultiError()
```

Comment:
The valkyrie no longer supports constructor based initialization. We should change it.